### PR TITLE
DEVXP-2794 - Add Formatter json

### PR DESCRIPTION
**Pull request by vschoener**
I've added an easy `Protocol` implementation to handle data for each dependency iteration.
This could be done differently in a much better isolated way but for a first version, it could be sufficient.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# EditorConfig is awesome: https://EditorConfig.org
```
Comment (vschoener) [inline .editorconfig:1]: Bonus

File: libyear/export_format/ascii.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from prettytable import PrettyTable
+
+from libyear.export_format.export_format import Item
+from libyear.utils import calculate_libyear
+
+
+class ASCIIFormatter:
+    pt: PrettyTable
+    sort: str
```
Comment (vitorquintanilha) [inline libyear/export_format/ascii.py]: Is this used?

File: libyear/export_format/export_format.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+class Item(BaseModel):
+    name: str
+    version: str
+    latest_version: str
+    libyear: str
+
+
+class ExportFormat(Protocol):
+    def initialize(self, sort: str | None):
```
Comment (vitorquintanilha) [inline libyear/export_format/export_format.py]: I think on `ASCIIFormatter` we have `sort: bool`, and here we have `sort: str`.

File: libyear/libyear [hunk 4]
```diff
@@ -15,39 +14,37 @@ def main():
...
+            formatter.add_item(item=item)
 
-    if args.sort:
-        pt.sortby = 'Libyears behind'
-        pt.reversesort = True
+        total_days += days
 
     if total_days == 0:
-        logger.info("Your system is up-to-date!")
+        print("Your system is up-to-date!")
```
Comment (vitorquintanilha) [inline libyear/libyear]: Any reason for this change? ^^

File: pyproject.toml [hunk 5]
```diff
@@ -14,6 +14,7 @@ prettytable = "^3.10.0"
 python-dateutil = "^2.9.0.post0"
 setuptools = "^69.2.0"
 packaging = "^24.0"
+pydantic = "^2.6.4"
```
Comment (vitorquintanilha) [inline pyproject.toml:17]: Maybe that's a personal opinion, but if we are not using any pydantic-specific features we could rely on Python's built-in `dataclass` and avoid having an extra dependency.

File: libyear/export_format/export_format.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        """
+        pass
+
+    def end(self, days: int):
+        """
+        Use to set data after we iterate over the packages
+        """
+        pass
+
+    def print(self):
```
Comment (vitorquintanilha) [inline libyear/export_format/export_format.py]: To make it easier to test and more flexible on the future (e.g. if we want to print to a file), we could have `def print(self, output=typingTextIO)`, so you can define where you are printing into

File: libyear/export_format/export_format.py (hunk 3 above)
Comment (vschoener) [reply libyear/export_format/export_format.py]: Yes at the very beginning I used sort Asc and Desc but I decided to move back for simplicity. There are some leftover

File: libyear/libyear (hunk 4 above)
Comment (vschoener) [reply libyear/libyear]: Yes the test are falling if we log instead of print as it's based on this message as well

Comment (vschoener) [reply libyear/libyear]: I mean this specific line does not need to be changed to print but it was more for consistency with the table. I think this one can move back to log anyway

File: pyproject.toml (hunk 5 above)
Comment (vschoener) [reply pyproject.toml:17]: My thoughts on that are more:
- Having an extra package is not costly
- Pydantic feels consistent with other micro services/lib usage so if we can stick to one usage for every, that would be easier and nice to work with project

Let me know if it makes sense, I don't have strong opinion on using one or the other

File: libyear/export_format/json.py [hunk 7]
```diff
@@ -1,17 +1,45 @@
...
+                libyear=item.libyear
+            )
+        )
 
     def end(self, days: int):
-        pass
+        self.json_output.libyears_behind = days
+
+        if self.sort:
+            self.json_output.dependencies.sort(key= lambda dep : (dep.libyear, dep.name))
```
Comment (vschoener) [inline libyear/export_format/json.py:42]: I sort by `libyear` and also `name` to list it properly and also to avoid flaky tests at the same time :)

**Review by vitorquintanilha — APPROVED** [review]

**Pull request merged**